src/main/jjtree/bsh.jjt

```diff
@@ -775,62 +775,164 @@ int AssignmentOperator() :
     }
 }
 
+/*
+ * Here follows the productions for Java expressions.
+ * JavaCC implements operator precedence using the cascade of
+ * productions.  It is important to maintain these as separate
+ * independent productions in the reverse sequence of the order of
+ * precedence. Do not flatten these in to fewer productions in an
+ * attempt to make the parser go faster because:
+ * - it will lose all the BEMDAS/PEMDAS order of operations.
+ * - it probably wont go faster.  Don't forget JavaCC will emit java
+ * code that still has all of the conditional tests.  All flattening
+ * will do is rearrange them in to a different order.  The java code 
+ * will be compiled, jitted and inlined by c2 so you wont even know
+ * what the performance impact is unless you do micro benchmarks.
+ */
```

Comment:
This comment should only shed light on what the code does, if it is unclear otherwise. It should not be the motivation for the change, that is what the PR and commit messages are for.